### PR TITLE
[WIP] Add support for multiple /pony requests

### LINKED ISSUE #13938: Pony plugin: Cope with multiple requests in one comment
**What would you like to be added**:

Currently the pony plugin will only search for the first pony request it got in a comment, i.E. the comment:

```
/pony template
/pony boo
```

Will result in a "Couldn't find a pony for that query" and no `boo` pony which is disappointing. Instead, the pony plugin should look for both a `template` and a `boo` pony and post both of them or an error for each.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by praveensastry**
Closes #13938

### CODE REVIEW

File: prow/plugins/pony/pony.go [hunk 1]
```diff
@@ -125,34 +125,37 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
...
-		resp, err := p.readPony(tag)
-		if err != nil {
-			log.WithError(err).Println("Failed to get a pony")
-			continue
+	for _, mat := range mats {
+		qry := mat[0]
+		tag := mat[1]
+		org := e.Repo.Owner.Login
+		repo := e.Repo.Name
+		number := e.Number
```
Comment (alvaroaleman) [inline prow/plugins/pony/pony.go]: These three can be defined outside the loop

File: prow/plugins/pony/pony.go [hunk 2]
```diff
@@ -125,34 +125,37 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
 		return nil
 	}
 	// Make sure they are requesting a pony
-	mat := match.FindStringSubmatch(e.Body)
-	if mat == nil {
+	mats := match.FindAllStringSubmatch(e.Body)
```
Comment (alvaroaleman) [inline prow/plugins/pony/pony.go]: Maybe rename to `matches` for readability?

File: prow/plugins/pony/pony.go [hunk 3]
```diff
@@ -125,34 +125,37 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
...
-	org := e.Repo.Owner.Login
-	repo := e.Repo.Name
-	number := e.Number
-
-	for i := 0; i < 5; i++ {
-		resp, err := p.readPony(tag)
-		if err != nil {
-			log.WithError(err).Println("Failed to get a pony")
-			continue
+	for _, mat := range mats {
```
Comment (alvaroaleman) [inline prow/plugins/pony/pony.go]: Can we have a limit for the maximum number of pony requests per comment?

Comment (praveensastry) [reply prow/plugins/pony/pony.go]: Thank you for your comments @alvaroaleman, What is the right number for max ponies in a comment, I was thinking 5. Should I go ahead with 5?

Comment (alvaroaleman) [reply prow/plugins/pony/pony.go]: Five sounds good to me

**Comment by matthyx** [comment]
@praveensastry do you plan to come back to this?

**Comment by matthyx** [comment]
/pony sad
@praveensastry any chance to get back to this?

**Comment by praveensastry** [comment]
@matthyx I will try and finish it this weekend. I was on a sabbatical and couldn’t get much time to work on this.

**Pull request closed without merging**